### PR TITLE
feat: add validation zod for getDistance & orderEstimation

### PR DESCRIPTION
**Pull request by hanggarHungry24**
ticket basecamp :  

https://3.basecamp.com/5190892/buckets/31126446/todos/5907615957     https://3.basecamp.com/5190892/buckets/31126446/todos/5907622456

### CODE REVIEW

File: api/book/getDistance.ts [hunk 1]
```diff
@@ -32,15 +33,20 @@ export async function getDistance(
...
     const responseData = data as z.infer<typeof getDistanceResponseSchema>;
     return {
       success: responseData.success,
       message: responseData.message,
       data: responseData.data,
     };
   } catch (err) {
-    const defaultErrorMessage = `${DEFAULT_ERROR_MESSAGE}, failed get disance`;
+    const defaultErrorMessage = `${DEFAULT_ERROR_MESSAGE}, failed to get distance`;
+    useReport({
```
Comment (hanggarHungry24) [inline api/book/getDistance.ts:45]: im not sure is this the correct way to use useReport mas @afif-hh  ?

File: api/book/getDistance.ts [hunk 2]
```diff
@@ -1,54 +1,72 @@
...
+import { useReport } from "~~/composables/useReport";
+const getDistancePayloadSchema = z.object({
+  restaurantId: z.union([z.string(), z.number()]),
+  lat: z.number(),
+  lon: z.number(),
+});
+export const getDistanceAttrsSchema = z
+  .object({
+    distance: z.number(),
+    min_distance: z.number(),
```
Comment (afif-hh) [inline api/book/getDistance.ts:15]: pakai camelCase mas

File: api/book/getDistance.ts [hunk 3]
```diff
@@ -1,54 +1,72 @@
...
-
-    if (error?.message || httpStatus !== 200) {
-      return {
-        success: false,
-        message: error.message || defaultErrorMessage,
-        data: null,
-      };
-    }
-
+    const responseData = data as z.infer<typeof getDistanceResponseSchema>;
```
Comment (afif-hh) [inline api/book/getDistance.ts:37]: respon nya di parse juga mas pakai zod, jadi bakal error kalau ga sesuai

File: api/book/getDistance.ts (hunk 1 above)
Comment (afif-hh) [reply api/book/getDistance.ts:45]: udah bener mas, `if` yg bawah" belum di useReport

**Review by afif-hh — CHANGES_REQUESTED** [review]

File: api/book/getDistance.ts [hunk 4]
```diff
@@ -1,54 +1,68 @@
...
     return {
-      success: responseData.success,
-      message: responseData.message,
-      data: responseData.data,
+      success,
+      message,
+      data,
     };
   } catch (err) {
+    const defaultErrorMessage = `${DEFAULT_ERROR_MESSAGE}, failed to get distance`;
```
Comment (afif-hh) [inline api/book/getDistance.ts:47]: bisa pakai useHandleError kemarin mas

File: api/book/orderTimeEstimation.ts [hunk 5]
```diff
@@ -1,56 +1,76 @@
...
     return {
-      success: responseData.success,
-      message: responseData.message,
-      data: responseData.data,
+      success,
+      message,
+      data,
     };
   } catch (err) {
+    const defaultErrorMessage = `${DEFAULT_ERROR_MESSAGE}, failed to get estimation data`;
```
Comment (afif-hh) [inline api/book/orderTimeEstimation.ts:55]: semua pakai useHandleError mas di catch nya

**Review by afif-hh — CHANGES_REQUESTED** [review]

File: api/book/getDistance.ts (hunk 4 above)
Comment (hanggarHungry24) [reply api/book/getDistance.ts:45]: udah di update mas

**Review by afif-hh — APPROVED** [review]

**Pull request merged**